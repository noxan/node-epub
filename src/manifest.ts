import camelcaseKeys from 'camelcase-keys';

export const parseManifest = (manifest: any) => {
  const result: any = {};

  manifest.item.forEach((item: any) => {
    result[item.id] = camelcaseKeys(item);
  });

  return result;
};
