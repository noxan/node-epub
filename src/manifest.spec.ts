import fixutureMini from './manifest.fixture-mini.json';
import fixutureFull from './manifest.fixture-full.json';
import { parseManifest } from './manifest';

describe('manifest', () => {
  it('transform manifest minimal fixture into key based object', () => {
    const manifest = parseManifest(fixutureMini);
    expect(manifest).toEqual({
      cover: {
        href: 'cover.xhtml',
        id: 'cover',
        mediaType: 'application/xhtml+xml',
      },
    });
  });

  it('transform manifest full fixture into key based object', () => {
    const manifest = parseManifest(fixutureFull);
    expect(manifest).toMatchSnapshot();
  });
});
