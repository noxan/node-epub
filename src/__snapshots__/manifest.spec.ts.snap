// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manifest transform manifest full fixture into key based object 1`] = `
Object {
  "chapter-1": Object {
    "href": "chapter-1.xhtml",
    "id": "chapter-1",
    "mediaType": "application/xhtml+xml",
    "properties": "scripted",
  },
  "cover": Object {
    "href": "cover.xhtml",
    "id": "cover",
    "mediaType": "application/xhtml+xml",
  },
  "cover-image": Object {
    "href": "images/cover-image.png",
    "id": "cover-image",
    "mediaType": "image/png",
  },
  "javascript": Object {
    "href": "js/book.js",
    "id": "javascript",
    "mediaType": "text/javascript",
  },
  "ncx": Object {
    "href": "epb.ncx",
    "id": "ncx",
    "mediaType": "application/x-dtbncx+xml",
  },
  "stylesheet": Object {
    "href": "css/book.css",
    "id": "stylesheet",
    "mediaType": "text/css",
  },
  "toc": Object {
    "href": "toc.xhtml",
    "id": "toc",
    "mediaType": "application/xhtml+xml",
    "properties": "nav",
  },
}
`;
